pkg/pillar/types/errortime.go

```diff
@@ -34,36 +108,43 @@ func (etPtr *ErrorAndTime) SetError(errStr string, errorTime time.Time) {
 	}
 	etPtr.Error = errStr
 	etPtr.ErrorTime = errorTime
+	etPtr.ErrorSeverity = ErrorSeverityError
+}
+
+// SetErrorWithDescription is when time and description are specified
+// it sets ErrorSeverityError in case of unspecified ErrorSeverity
+func (etPtr *ErrorAndTime) SetErrorWithDescription(errDescription ErrorDescription, errorTime time.Time) {
```

Comment:
If we fold errorTime into ErrorDescription then we can have this function check if errorTime is zero and if so set it to time.Now().

Reply:
Moved